### PR TITLE
Add devise gem

### PR DESCRIPTION
**Pull request by juan-apa**
- Added devise gem
- Created sign-up feature @ POST /api/v1/users
- Added circleci config files

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
   # Store uploaded files on the local file system (see config/storage.yml for options).
   config.active_storage.service = :local
-
+  config.action_mailer.default_url_options = { host: 'dev.yourhost.com' }
```
Comment (mdornel) [inline config/environments/development.rb]: Remove this line and add something like this in `config/application.rb`:
```
config.action_mailer.default_url_options = { host: ENV['SERVER_URL'] }
```

This way you avoid to add this configuration on all your environments, then you'll have just to modify `SERVER_URL` value for each environment.

Check out [this](https://guides.rubyonrails.org/configuring.html) for more info about configuring you app

File: config/environments/test.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@
   # just for the purpose of running a single test. If you are using a tool that
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
+  config.action_mailer.default_url_options = { host: 'test.yourhost.com' }
```
Comment (mdornel) [inline config/environments/test.rb]: Same here

File: .circleci/application.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+SERVER_URL: 'localhost:3000'
+PASSWORD_RESET_URL: 'http://localhost:3000'
```
Comment (sebaherrera07) [inline .circleci/application.yml]: Missing empty line at the end.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      # run code_analysis!
+      - run:
+          name: run rails code analysis
+          command: RAILS_ENV=test bundle exec rake code_analysis
+
+      - store_test_results:
+          path: /tmp/test-results
+      - store_artifacts:
+          path: /tmp/test-results
+          destination: test-results
```
Comment (sebaherrera07) [inline .circleci/config.yml]: Missing empty line at the end.

File: .circleci/database.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+test:
+  adapter: postgresql
+  encoding: unicode
+  pool: 5
+  username: postgres
+  password: postgres
+  host: localhost
+  port: 5432
+  database: test_db
```
Comment (sebaherrera07) [inline .circleci/database.yml]: Missing empty line at the end.

**Pull request merged**